index.js

```diff
@@ -0,0 +1,57 @@
+const express = require('express');
+const cors = require('cors');
+
+const bcrypt = require('bcryptjs');
+
+const db = require('./database/dbConfig.js')
+
+const server = express();
+
+const PORT = 5050;
+
+server.use(express.json());
+server.use(cors());
+
+
+server.post('/api/register', (req, res) =>{
+    const user = req.body;
+    const hash = bcrypt.hashSync(user.password, 14);
+    user.password = hash;
+    if(!user.username || !user.password){res.status(400).json({message:'Please provide username and password!'})}
+    db.insertUser(user)
+    .then(ids =>{
+        const id = ids[0]
```

Comment:
There are a couple of alternative ways to get that `id` off of the array:
- [Array destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring):
```javascript
db.insertUser(user)
  .then(([ id ]) => {
    res.json({ newUserId: id })
  })
  // ...
```
- the [first](https://knexjs.org/#Builder-first) method
```javascript
db.insertUser(user)
  .first()
  .then(id => {
    res.json({ newUserId: id })
  })
  // ...
```